NRF52 : Fix UART RTS initialization

naveenkaje opened the pull request:
The pre-processor based macro check #if evaluates all enums as
0 and hence the code within #if does not get compiled. Hence
move this to a run-time check where the pin variable can be
correctly evaluated.

This is a follow on patch to:
https://github.com/ARMmbed/mbed-os/pull/8046
The above PR does not work for NRF52840_DK and needs this fix.

    [x] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

Commit:
NRF52 : Fix UART RTS initialization

The preprocessor based macro check #if evaluates all
enums as 0 and hence the code does not get compiled.
Since move this to a runtime check where the pin variable
can be correctly evaluated.

Delete mbed_overrides.c as it has a target specific mbed_sdk_init() to
resolve linking problem.

This is a follow on patch to:
https://github.com/ARMmbed/mbed-os/pull/8046

naveenkaje commented:
@SenRamakri @c1728p9

cmonr commented:
> This breaks NRF52840_DK and needs this fix.

The first 'this'. Do you mean the enum preprocessor issue?

naveenkaje commented:
@cmonr Sorry for the cryptic message. I changed it to "The above PR does not work for NRF52840_DK and needs this fix."

0xc0170 commented:
LGTM (cant approve as github  is having issues), will start CI

/morph build

0xc0170 commented:
LGTM (cant approve as github  is having issues), will start CI

/morph build

cmonr commented:
@naveenkaje Was this a rebase, or were the failure issues fixed and squashed? Hard to tell.

naveenkaje commented:
@cmonr I had originally changed only targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/reloc_vector_table.c, but during CI I noted that the following files needed the RTS pin definition

targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/TARGET_MCU_NRF52832/TARGET_DELTA_DFBM_NQ620/PinNames.h
targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/TARGET_MCU_NRF52832/TARGET_MTB_UBLOX_NINA_B1/PinNames.h
targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/TARGET_MCU_NRF52832/TARGET_RBLAB_BLENANO2/PinNames.h
targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/TARGET_MCU_NRF52832/TARGET_UBLOX_EVA_NINA/PinNames.h

Hence I modified them and rebased on top of master.
Hope it's clear.

cmonr commented:
/morph build

cmonr commented:
Interesting. Looks like the last build issue is only with the DELTA_DFBM_NQ620 target.

naveenkaje commented:
Thanks @cmonr 

It is caused due to multiply defined mbed_sdk_init() for that target.
> TESTS/mbed_platform/system_reset/main.cpp
        [DEBUG] Return: 0
        Link: system_reset_application
        Link: /usr/local/ARM_Compiler_5.06u3/bin/armlink --via /builds/ws/mbed-os-build-matrix/3430/DELTA_DFBM_NQ620_ARM/sources/mbed-os/BUILD/tests/DELTA_DFBM_NQ620/ARM/TESTS/mbed_platform/system_reset/.link_options.txt
        [DEBUG] Return: 1
        [DEBUG] Errors: Error: L6200E: Symbol mbed_sdk_init multiply defined (by BUILD/tests/DELTA_DFBM_NQ620/ARM/targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/reloc_vector_table.o and BUILD/tests/DELTA_DFBM_NQ620/ARM/targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/TARGET_MCU_NRF52832/TARGET_DELTA_DFBM_NQ620/mbed_overrides.o).

Looking at the history, 56cf0921adf8c4c8f00a57f4ee8e383fadd30f55 has brought this call for that target. The function is not doing anything particularly useful. It is declaring a couple of local variables. I propose to delete this file.

cmonr commented:
/morph build

cmonr commented:
Job config fixed.
/morph test

cmonr commented:
Job config fixed.
/morph test

cmonr commented:
Waiting on new greentea tests. Should be about an hour until complete.
